simplified-cardcreator/src/main/java/org/nypl/simplified/cardcreator/ui/JuvenilePolicyFragment.kt

```diff
@@ -95,129 +92,98 @@ class JuvenilePolicyFragment : Fragment() {
     // Go to next screen
     binding.nextBtn.setOnClickListener {
       if (validateForm()) {
-        binding.progress.visibility = View.VISIBLE
-        getToken()
+        getEligibility()
       } else {
         requireActivity().setResult(Activity.RESULT_CANCELED)
         requireActivity().finish()
       }
     }
 
-    viewModel.issoTokenData.observe(
-      viewLifecycleOwner,
-      Observer {
-        getCache().token = it.access_token
-        getEligibility()
-      }
-    )
+    val callback = requireActivity().onBackPressedDispatcher.addCallback(this) {
+      requireActivity().setResult(Activity.RESULT_CANCELED)
+      requireActivity().finish()
+    }
+    callback.isEnabled = true
 
-    platformViewModel.dependentEligibilityData.observe(
-      viewLifecycleOwner,
-      Observer {
-        if (it.eligible) {
-          nextAction = JuvenilePolicyFragmentDirections.actionLocation()
-          navController.navigate(nextAction)
-        } else {
-          val dialogBuilder = AlertDialog.Builder(requireContext())
-          dialogBuilder.setMessage(it.description)
-            .setCancelable(false)
-            .setNegativeButton(getString(R.string.cancel)) { _, _ ->
-              requireActivity().setResult(Activity.RESULT_CANCELED)
-              requireActivity().finish()
-            }
-          if (dialog == null) {
-            dialog = dialogBuilder.create()
-          }
-          dialog?.show()
-        }
-      }
-    )
+    viewModel.dependentEligibilityResponse
+      .receive(viewLifecycleOwner, this::handleDependentEligibilityResponse)
 
-    platformViewModel.apiErrorMessage.observe(
-      viewLifecycleOwner,
-      Observer {
-        binding.progress.visibility = View.GONE
-        val dialogBuilder = AlertDialog.Builder(requireContext())
-        dialogBuilder.setMessage("$it")
-          .setCancelable(false)
-          .setPositiveButton(getString(R.string.try_again)) { _, _ ->
-            getToken()
-          }
-          .setNegativeButton(getString(R.string.cancel)) { dialog, _ ->
-            dialog.cancel()
-          }
-        if (dialog == null) {
-          dialog = dialogBuilder.create()
-        }
-        dialog?.show()
-      }
-    )
+    viewModel.pendingRequest.observe(viewLifecycleOwner, this::showLoading)
+  }
 
-    platformViewModel.apiError.observe(
-      viewLifecycleOwner,
-      Observer {
-        binding.progress.visibility = View.GONE
-        val dialogBuilder = AlertDialog.Builder(requireContext())
-        dialogBuilder.setMessage("$it Error")
-          .setCancelable(false)
-          .setPositiveButton(getString(R.string.try_again)) { _, _ ->
-            getToken()
+  private fun showLoading(loading: Boolean) {
+    if (loading) {
+      binding.progress.visibility = View.VISIBLE
+    } else {
+      binding.progress.visibility = View.GONE
+    }
```

Comment:
This can be simplified with the Kotlin core method by using `binding.progress.isVisible = loading`

Reply:
Another good call, but same as above. The conflicts are a real pain when I merge in upstream changes.